bon-macros/src/lib.rs

```diff
@@ -100,3 +102,93 @@ pub fn bon(params: TokenStream, item: TokenStream) -> TokenStream {
         .unwrap_or_else(|err| error::error_into_token_stream(err, item.into()))
         .into()
 }
+
+/// Similar to [`maplit::hashmap!`]/[`maplit::btreemap!`] but converts each key and value with
+/// [`Into::into()`].
+///
+/// There are no separate variants for [`HashMap`] and [`BTreeMap`] since the macro works with any
+/// type that implements [`FromIterator<(K, V)>`].
+///
+/// The macro also performs rudimentary uniqueness checking on keys: Syntactically equal keys are
+/// rejected with a compile error.
+///
+/// A good example of the use case for this macro is when you want to create a
+/// `HashMap<String, String>` where part of the keys or values are hardcoded string literals of type `&str`
+/// and the other part is made of dynamic [`String`] values.
+///
+/// ```rust
+/// # use bon_macros as bon;
+/// # use std::collections::HashMap;
+/// fn convert_media(input_extension: &str, output_extension: &str) -> std::io::Result<()> {
+///     let ffmpeg_args: HashMap<String, String> = bon::map!{
+///         "-i": format!("input.{input_extension}"),
+///         "-y": format!("output.{output_extension}"),
+///     };
+///
+///     let mut command = std::process::Command::new("ffmpeg");
+///     for (flag, value) in ffmpeg_args {
+///         command.arg(flag).arg(value);
+///     }
+///     command.output()?;
```

Comment:
There are a couple of problems with this example.

First, it should showcase that both the keys of the map can have diverse types and values too. For example:

```rust
let map: HashMap<String, String> = bon::map! { 
    "&str literal key": format!("Owned string value: {}", 2 + 2),
    format!("Owned String key"): "&str literal value",
};
```

I can't give an immediate good real-world example to showcase this, but the current example with `ffmpeg` CLI doesn't show this feature.

The second problem is a bit of a logical bug. The flag `-y` in `ffmpeg` is  a boolean fla. From the `ffmpeg` docs: (https://www.ffmpeg.org/ffmpeg.html)):
<img src="https://github.com/user-attachments/assets/f5854169-7559-4339-a2c1-308bcbc862bb" width="300px"/>

So it doesn't take any value, and it isn't correct to pair it with the output file path in the map key-value pair.

If you can't think of any other good real-world use case for a map with some keys of string literal types, and some of dynamic owned strings, then let's just use a dummy example that uses `foo`s and `bar`s as values. We can update the docs with a better example any time anyway.

Reply:
I did a mostly synthetic example that still shows the macro in action.

I could also split the example, showing the working code first and a compile fail second.

Reply:
Let's probably avoid using `ffmpeg` in the examples. Let's make the either fully real-world, or fully synthetic. It could look like a unit test that just creates a map and asserts that it contains the expected keys. And yeah, a separate example for the validation of uniqueness would be cool